framework/src/modules/chain/loader.js

```diff
@@ -385,24 +355,49 @@ class Loader {
 		let loaded = false;
 		while (!loaded && failedAttemptsToLoad < 5) {
 			try {
-				// eslint-disable-next-line no-await-in-loop
-				const blocksFromNetwork = await this._getBlocksFromNetwork();
-				// eslint-disable-next-line no-await-in-loop
-				const blocksAfterValidate = await this._validateBlocks(
-					blocksFromNetwork,
-				);
+				let blocksAfterValidate;
+				try {
+					// eslint-disable-next-line no-await-in-loop
+					const blocksFromNetwork = await this._getBlocksFromNetwork();
+					// eslint-disable-next-line no-await-in-loop
+					blocksAfterValidate = await this._validateBlocks(blocksFromNetwork);
+				} catch (loadBlocksFromNetworkErr) {
+					// eslint-disable-next-line no-await-in-loop
+					await this._handleBlocksFromNetworkError(loadBlocksFromNetworkErr);
+				}
 				// eslint-disable-next-line no-await-in-loop
 				loaded = await this._getValidatedBlocksFromNetwork(blocksAfterValidate);
 				// Reset counter after a batch of blocks was successfully loaded from the network
 				failedAttemptsToLoad = 0;
 			} catch (err) {
-				if (err) {
-					failedAttemptsToLoad += 1;
-					this.logger.error(convertErrorsToString(err));
-				}
+				failedAttemptsToLoad += 1;
+				this.logger.error(convertErrorsToString(err));
 			}
 		}
 	}
+
+	async _handleBlocksFromNetworkError(error) {
```

Comment:
This function is triggering chain recovery, the name is confusing and doesn't reflect the behavior. Also, with this logic, we have nested try-catch block (3 levels). Can we refactor this in a better way?

Reply:
yes, i found a bug in this way, so im gonna clean it up